src/LtiAssignmentsGradesService.php

```diff
@@ -2,75 +2,64 @@
-        $score_url = $pos === false ? $score_url.'/scores' : substr_replace($score_url, '/scores', $pos, 0);
-
-        return $this->service_connector->makeServiceRequest(
-            $this->service_data['scope'],
-            LtiServiceConnector::METHOD_POST,
-            $score_url,
-            $grade,
-            'application/vnd.ims.lis.v1.score+json'
-        );
+        $pos = strpos($scoreUrl, '?');
+        $scoreUrl = $pos === false ? $scoreUrl.'/scores' : substr_replace($scoreUrl, '/scores', $pos, 0);
+
+        $request = new ServiceRequest(LtiServiceConnector::METHOD_POST, $scoreUrl);
+        $request->setBody($grade);
+        $request->setContentType('application/vnd.ims.lis.v1.score+json');
+
+        return $this->makeServiceRequest($request);
     }
 
-    public function findOrCreateLineitem(LtiLineitem $new_line_item)
+    public function findOrCreateLineitem(LtiLineitem $newLineItem)
     {
-        $line_items = $this->getLineItems();
+        $lineitems = $this->getLineItems();
 
-        foreach ($line_items as $line_item) {
+        foreach ($lineitems as $lineitem) {
             if (
-                (empty($new_line_item->getResourceId()) && empty($new_line_item->getResourceLinkId())) ||
-                (isset($line_item['resourceId']) && $line_item['resourceId'] == $new_line_item->getResourceId()) ||
-                (isset($line_item['resourceLinkId']) && $line_item['resourceLinkId'] == $new_line_item->getResourceLinkId())
+                (empty($newLineItem->getResourceId()) && empty($newLineItem->getResourceLinkId())) ||
+                (isset($lineitem['resourceId']) && $lineitem['resourceId'] == $newLineItem->getResourceId()) ||
+                (isset($lineitem['resourceLinkId']) && $lineitem['resourceLinkId'] == $newLineItem->getResourceLinkId())
             ) {
-                if (empty($new_line_item->getTag()) || $line_item['tag'] == $new_line_item->getTag()) {
-                    return new LtiLineitem($line_item);
+                if (empty($newLineItem->getTag()) || $lineitem['tag'] == $newLineItem->getTag()) {
+                    return new LtiLineitem($lineitem);
                 }
             }
         }
-        $created_line_item = $this->service_connector->makeServiceRequest(
-            $this->service_data['scope'],
-            LtiServiceConnector::METHOD_POST,
-            $this->service_data['lineitems'],
-            $new_line_item,
-            'application/vnd.ims.lis.v2.lineitem+json',
-            'application/vnd.ims.lis.v2.lineitem+json'
-        );
-
-        return new LtiLineitem($created_line_item['body']);
+        $request = new ServiceRequest(LtiServiceConnector::METHOD_POST, $this->getServiceData()['lineitems']);
+        $request->setBody($newLineItem)
+            ->setContentType('application/vnd.ims.lis.v2.lineitem+json')
+            ->setAccept('application/vnd.ims.lis.v2.lineitem+json');
+        $created_lineitem = $this->makeServiceRequest($request);
```

Comment:
Can we camel case `$createdLineItems` ?

```suggestion
        $createdLineItems = $this->makeServiceRequest($request);
```